tests/models/test_dagrun.py

```diff
@@ -234,8 +234,9 @@ def test_dagrun_deadlock(self):
         ti_op2.set_state(state=State.NONE, session=session)
 
         dr.update_state()
-        self.assertEqual(dr.state, State.RUNNING)
+        self.assertEqual(dr.state, State.FAILED)
 
+        dr.set_state(State.RUNNING)
```

Comment:
So before your change these tasks ended up in this state: `[<TaskInstance: text_dagrun_deadlock.B [skipped]>, <TaskInstance: text_dagrun_deadlock.A [success]>]`. And looking at the TriggerRuleDep for ONE_FAILED:

```
            elif tr == TR.ONE_FAILED:
                if upstream_done and not (failed or upstream_failed):
                    ti.set_state(State.SKIPPED, session)
```

So that looks right - op2/B should be in SKIPPED state.

@amichai07 So this is a bug, and the test was right and shouldn't be changed.

Reply:
The only thing I wondered about was whether it should be "upstream_failed" instead of "skipped"